docs/user-guide/optimization.md

```diff
@@ -0,0 +1,109 @@
+# Optimization
+
+Cubed will automatically optimize the computation graph before running it. This can reduce the number of tasks in the plan, and the amount of intermediate IO, both of which speed up the computation.
+
+## Map fusion
+
+The simplest kind of optimization is _map fusion_, where operations that have one preceding operation with the same number of tasks are fused together. This optimization is enabled by default.
+
+You can see the effect of optimization before running any computation by using the `visualize` method on a Cubed array, such as in the following small example. We start by specifying `optimize_graph=False` to turn off optimization so we can see what the unoptimized plan looks like.
+
+```python
+import cubed.array_api as xp
+
+a = xp.asarray([[1, 2, 3], [4, 5, 6], [7, 8, 9]], chunks=(2, 2))
+b = xp.negative(a)
+c = xp.astype(b, xp.float32)
+
+c.visualize("cubed-unoptimized", optimize_graph=False)
+```
+
+![Computation with optimization turned off](../images/optimization_turned_off.svg)
+
+Now we call `visualize` again, this time not setting `optimize_graph` so it picks up its default value of `True`.
+
+```python
+c.visualize()
+```
+
+![Map fusion optimization](../images/optimization_map_fusion.svg)
+
+Note that with optimization turned on, the array `b` is no longer written as an intermediate output since it will be computed in the same tasks that compute array `c`. The overall number of tasks is reduced from 10 to 5, and the intermediate data (total nbytes) is reduced too.
```

Comment:
Maybe note that this optimization is always beneficial, which is why it's on by default?